src/pcre2_compile.c

```diff
@@ -8794,10 +8794,10 @@ for (;;)
     {
     if (lengthptr == NULL)
       {
-      PCRE2_SIZE branch_length = code - last_branch;
+      uint32_t branch_length = (uint32_t)(code - last_branch);
```

Comment:
Don't change the type. Just add the casting.

Reply:
there is no point having a PCRE2_SIZE variable here if the data we are going to handle is always going to be less than UINT32_MAX.

Additionally, we have to `PUT` this value below and that would also require a casting in that case.

Reply:
`PUT` needs uint32_t? So far no compiler is complained about this before.

Reply:
An example [run](https://github.com/carenas/pcre2/actions/runs/10916668606/job/30298529257) with the proposed change and without the cast in PUT.

Reply:
https://github.com/PCRE2Project/pcre2/blob/master/src/pcre2_intmodedep.h#L105

The macro has an explicit type cast. Even with `uint32_t`, it is often shortened to 8/16 bit. Why the compiler complains only here, and not in other similar cases?

Reply:
My guess (as explained before with clang) is that that specific truncation is common enough with the migration to 64bit CPUs to warrant a warning.

Reply:
But why only here? With explicit type casts? I suspect something is wrong with the macro definition, and should be fixed elsewhere. That is why I pushing the topic.

Reply:
Maybe clang could be used to get a more detailed info about the implicit type cast which we don't see. If it shows as detailed info as `gcc` these days.

Reply:
Again, I think that the difference comes from the fact that truncating a 64bit integer into 32bits (which is what PCRE2_UCHAR is for the 32bit library) triggers these warnings (by default in MSVC and with extra help in clang).

That doesn't mean that truncating it to 16 o 8bit isn't as problematic, but the compiler writers probably wanted to focus on the ones that will be more likely in current codebases.

FWIW, there are several PUT invocations already with casts, which might had been added because of similar warnings, but at least in this case, the best will be to avoid the cast and use a type of the right size, so it will be IMHO better to resolve this specific thread.